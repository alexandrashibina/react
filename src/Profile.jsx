import React, {Component} from 'react';
import {connect} from 'react-redux';
import { addBankCard } from './actions';
import mc from './mc.svg';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


export class Profile extends Component {
  
  handleSubmit = (event) => {
    event.preventDefault();
    const { cardNumber, expiryDate, cardName, cvc } = event.target;
    this.props.addBankCard(cardNumber.value, expiryDate.value, cardName.value, cvc.value);
  }

  render() {
    return (
      <>
        {this.props.cardAdded ? (
          <div data-testid="profile" className="container"> 
          <div className="login">
            <div className="form">
              <div className="profile__content">
              <div className="form__header profile">Профиль</div>
              <div className="form__text profile">Способ оплаты</div>
              </div>
              <form className="form__input profile__input">
                <div className="form__input-container">
                  <div className="column">
                    Спасибо, ваши платежные данные сохранены!
                  </div>
                </div>
                <Link to="map" className="button">Построить маршрут</Link>
              </form>
            </div>
          </div>
        </div>
        ) : (
        <div data-testid="profile" className="container"> 
          <div className="login">
            <div className="form">
              <div className="profile__content">
              <div className="form__header profile">Профиль</div>
              <div className="form__text profile">Способ оплаты</div>
              </div>
              <form onSubmit={this.handleSubmit} className="form__input profile__input">
                <div className="form__input-container">
                  <div className="column">
                  <span className="mc-icon"><img className="mc-img" src={mc} alt="MC"/></span>
                    <input name="cardNumber" className="input" type="text" placeholder="Номер карты"/>
                    <input name="expiryDate" className="input" type="text" placeholder="01/21"/>
                  </div>
                  <div className="column">
                    <input name="cardName" className="input" type="text" placeholder="Имя владельца"/>
                    <input name="cvc" className="input" type="text" placeholder="CVC"/>
                  </div>
                </div>
                <button type="submit" to="map" className="button">Сохранить данные</button>
              </form>
            </div>
          </div>
        </div>
        )}
      </>
    );
  }
} 

Profile.propTypes = {
  cardAdded: PropTypes.bool,
};

export const ProfileWithConnect = connect(
  (state) => ({cardAdded: state.card.cardAdded}),
  {addBankCard}
)(Profile);