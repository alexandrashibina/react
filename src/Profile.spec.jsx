import React from "react";
import { Profile } from "./Profile";
import renderer from "react-test-renderer";

describe("Profile", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
   });
})