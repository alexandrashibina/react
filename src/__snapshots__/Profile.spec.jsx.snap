// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="container"
  data-testid="profile"
>
  <div
    className="login"
  >
    <div
      className="form"
    >
      <div
        className="profile__content"
      >
        <div
          className="form__header profile"
        >
          Профиль
        </div>
        <div
          className="form__text profile"
        >
          Способ оплаты
        </div>
      </div>
      <form
        className="form__input profile__input"
        onSubmit={[Function]}
      >
        <div
          className="form__input-container"
        >
          <div
            className="column"
          >
            <span
              className="mc-icon"
            >
              <img
                alt="MC"
                className="mc-img"
                src="mc.svg"
              />
            </span>
            <input
              className="input"
              name="cardNumber"
              placeholder="Номер карты"
              type="text"
            />
            <input
              className="input"
              name="expiryDate"
              placeholder="01/21"
              type="text"
            />
          </div>
          <div
            className="column"
          >
            <input
              className="input"
              name="cardName"
              placeholder="Имя владельца"
              type="text"
            />
            <input
              className="input"
              name="cvc"
              placeholder="CVC"
              type="text"
            />
          </div>
        </div>
        <button
          className="button"
          to="map"
          type="submit"
        >
          Сохранить данные
        </button>
      </form>
    </div>
  </div>
</div>
`;
